### PR TITLE
[plugin] apply only valid configurations as schemas

### LINKED ISSUE #4621: [vscode] PreferenceSchemaProvider.setSchema: Cannot convert undefined or null to object
Reproducible with https://github.com/Microsoft/vscode-extension-samples/tree/master/tree-view-sample
```
root ERROR TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at PreferenceSchemaProvider.../../packages/core/lib/browser/preferences/preference-contribution.js.PreferenceSchemaProvider.doSetSchema (https://3000-b4bfab3e-b4e2-4935-9d1f-27cd3dc80676.ws-eu0.gitpod.io/bundle.js:83117:43)
    at PreferenceSchemaProvider.../../packages/core/lib/browser/preferences/preference-contribution.js.PreferenceSchemaProvider.setSchema (https://3000-b4bfab3e-b4e2-4935-9d1f-27cd3dc80676.ws-eu0.gitpod.io/bundle.js:83178:28)
    at PluginContributionHandler.push.../../packages/plugin-ext/lib/main/browser/plugin-contribution-handler.js.PluginContributionHandler.updateConfigurationSchema (https://3000-b4bfab3e-b4e2-4935-9d1f-27cd3dc80676.ws-eu0.gitpod.io/82.bundle.js:10401:39)
    at PluginContributionHandler.push.../../packages/plugin-ext/lib/main/browser/plugin-contribution-handler.js.PluginContributionHandler.handleContributions (https://3000-b4bfab3e-b4e2-4935-9d1f-27cd3dc80676.ws-eu0.gitpod.io/82.bundle.js:10227:18)
    at HostedPluginSupport.push.../../packages/plugin-ext/lib/hosted/browser/hosted-plugin.js.HostedPluginSupport.initContributions (https://3000-b4bfab3e-b4e2-4935-9d1f-27cd3dc80676.ws-eu0.gitpod.io/82.bundle.js:4379:46)
    at HostedPluginSupport.push.../../packages/plugin-ext/lib/hosted/browser/hosted-plugin.js.HostedPluginSupport.loadPlugins (https://3000-b4bfab3e-b4e2-4935-9d1f-27cd3dc80676.ws-eu0.gitpod.io/82.bundle.js:4313:30)
```

### PR DESCRIPTION
**Pull request by akosyakov**
fix #4621

in order to verify install https://github.com/akosyakov/vscode-trees VS Code extension:
```bash
mkdir plugins && /
cd plugins && /
git clone https://github.com/akosyakov/vscode-trees.git && /
cd vscode-trees && /
npm i && /
npm run compile
```
 and make sure that it loads `npm dependencies` tree

### CODE REVIEW

**Comment by benoitf** [comment]
cc @akurinnoy

File: packages/plugin-ext/src/common/plugin-protocol.ts [hunk 1]
```diff
@@ -55,8 +53,8 @@ export interface PluginPackage {
  * This interface describes a package.json contribution section object.
  */
 export interface PluginPackageContribution {
-    configuration?: PreferenceSchema;
-    configurationDefaults?: PreferenceSchemaProperties;
+    configuration?: Object;
```
Comment (benoitf) [inline packages/plugin-ext/src/common/plugin-protocol.ts]: I'm unsure the fix is to loose the Type ?

Comment (akosyakov) [reply packages/plugin-ext/src/common/plugin-protocol.ts]: in this case, the configuration in `package.json` is bogus, one has to validate that it confirms to `PreferenceSchema` before applying, the correct type would be something like `RecursivePartial<PreferenceSchema>`

File: packages/plugin-ext/src/hosted/node/scanners/scanner-theia.ts [hunk 2]
```diff
@@ -54,6 +54,8 @@ import * as jsoncparser from 'jsonc-parser';
 import { IJSONSchema } from '@theia/core/lib/common/json-schema';
 import { deepClone } from '@theia/core/lib/common/objects';
 import { FileUri } from '@theia/core/lib/node/file-uri';
+import { PreferenceSchema, PreferenceSchemaProperties } from '@theia/core/src/browser';
+import { RecursivePartial } from '@theia/core';
```
Comment (akosyakov) [inline packages/plugin-ext/src/hosted/node/scanners/scanner-theia.ts]: @theia/core/lib/common/types

File: packages/plugin-ext/src/hosted/node/scanners/scanner-theia.ts [hunk 3]
```diff
@@ -54,6 +54,8 @@ import * as jsoncparser from 'jsonc-parser';
 import { IJSONSchema } from '@theia/core/lib/common/json-schema';
 import { deepClone } from '@theia/core/lib/common/objects';
 import { FileUri } from '@theia/core/lib/node/file-uri';
+import { PreferenceSchema, PreferenceSchemaProperties } from '@theia/core/src/browser';
```
Comment (akosyakov) [inline packages/plugin-ext/src/hosted/node/scanners/scanner-theia.ts]: @theia/core/lib/common/preferences/preference-schema

File: packages/plugin-ext/src/common/plugin-protocol.ts (hunk 1 above)
Comment (akosyakov) [reply packages/plugin-ext/src/common/plugin-protocol.ts]: @benoitf fixed

**Pull request merged**